analyzer/analyzer.go

```diff
@@ -0,0 +1,48 @@
+package analyzer
+
+import (
+	"go/ast"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+// NewAnalyzer returns a new analyzer.
+func NewAnalyzer() *analysis.Analyzer {
+	return &analysis.Analyzer{
+		Name:     "standaloneFuncCalls",
+		Doc:      "Reports standalone function calls",
+		Run:      run,
+		Requires: []*analysis.Analyzer{inspect.Analyzer},
+	}
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	insp := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+
+	nodeFilter := []ast.Node{
+		(*ast.ExprStmt)(nil),     // to find expression statements
+		(*ast.CallExpr)(nil),     // to find call expressions
+		(*ast.SelectorExpr)(nil), // to find selector expressions
+	}
+
+	insp.Preorder(nodeFilter, func(n ast.Node) {
+		switch stmt := n.(type) {
+		case *ast.ExprStmt:
+			// Check if the expression is a call expression
+			switch expr := stmt.X.(type) {
+			case *ast.CallExpr:
+				// Report the standalone function call
+				pass.Reportf(expr.Pos(), "standalone function call found: %s", pass.Fset.Position(expr.Pos()))
+			case *ast.SelectorExpr:
+				// Check if the selector expression's X is a call expression
```

Comment:
I think the Go compiler will already forbid ineffectual selector expressions like

```go
type Struct struct {
    X int
    Y int
}


func NewStruct() Struct {
    return Struct{
        X: 1,
        Y: 2,
    }
}

func main() {
    NewStruct().X
}
```

Could you check on Go playground? If the compiler doesn't forbid it, let's keep the check, though I'd be very surprised if the compiler allowed it.